service/resource/cloudformationv2/adapter/adapter_test.go

```diff
@@ -24,28 +24,78 @@ var (
 )
 
 func TestAdapterGuestMain(t *testing.T) {
-	customObject := v1alpha1.AWSConfig{
-		Spec: v1alpha1.AWSConfigSpec{
-			Cluster: defaultCluster,
-			AWS: v1alpha1.AWSConfigSpecAWS{
-				AZ: "eu-central-1a",
-				Masters: []v1alpha1.AWSConfigSpecAWSNode{
-					v1alpha1.AWSConfigSpecAWSNode{
-						ImageID: "ami-test-master",
+	testCases := []struct {
+		description       string
+		customObject      v1alpha1.AWSConfig
+		errorMatcher      func(error) bool
+		expectedASGType   string
+		expectedClusterID string
+		expectedImageID   string
+	}{
+		{
+			description: "basic match",
+			customObject: v1alpha1.AWSConfig{
+				Spec: v1alpha1.AWSConfigSpec{
+					Cluster: defaultCluster,
+					AWS: v1alpha1.AWSConfigSpecAWS{
+						AZ:     "eu-central-1a",
+						Region: "eu-central-1",
+						Masters: []v1alpha1.AWSConfigSpecAWSNode{
+							v1alpha1.AWSConfigSpecAWSNode{},
+						},
+						Workers: []v1alpha1.AWSConfigSpecAWSNode{
+							v1alpha1.AWSConfigSpecAWSNode{},
+						},
 					},
 				},
-				Workers: []v1alpha1.AWSConfigSpecAWSNode{
-					v1alpha1.AWSConfigSpecAWSNode{
-						ImageID: "ami-test-worker",
+			},
+			errorMatcher:      nil,
+			expectedASGType:   "worker",
+			expectedClusterID: "test-cluster",
+			expectedImageID:   "ami-90c152ff",
```

Comment:
so this test needs to be updated with each CoreOS update, right? can we note that somewhere?